tlair/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/FlexibleEquality.scala

```diff
@@ -16,6 +17,8 @@ object FlexibleEquality {
   // None if no common supertype, Some(e) if compatible, and common supertype is e.
   def commonSupertype(lhs: TlaType1, rhs: TlaType1): Option[TlaType1] = (lhs, rhs) match {
     case (a, b) if a == b     => Some(a)
+    case (VarT1(_), b)        => Some(b)
+    case (a, VarT1(_))        => Some(a)
```

Comment:
@Kukovec I'd like to make sure you agree this is is the correct fix. This is overly permissive, as doesn't account for variable binding, but I think we should *not* put that logic in the builder, as we move much closer to reproducing the type checker in the builer then, which smells like tech debt, given that we can just use the type checker if we want :)

If you agree on this fix, then dozens of the builder test cases will need to be updated to account for type variability. 

My proposed fix, as seen below, is to clean up these complicated test cases by replacing them with simpler, PBT-style tests (which have -- or can be made to have -- equivalent or slightly better coverage, iiuc). These replacements use the concrete type builder introduced in #2550, which does not produce type variables.

Checks for logic around type variables can be confined to test of operator application, I believe.

WDYT?